### PR TITLE
To address the API review comments

### PR DESCRIPTION
**Pull request by saragluna**
To address the API review comments, this PR will: 
- Add the missing Javadocs
- Rename the `*OptionsAware` to `*OptionsProvider`
- Rename `Client`, `Retry` and others to `ClientOptions`, `RetryOptions` and `*Options`.
- Move the `headers` properties out from the `ClientOptions` and put it into the `HttpClientOptions` because
  > AMQP SDKs dismiss this configuration and yes, it might be misleading in SDK, but it would really be nice not to expose headers configuration for AMQP at this point (until SDKs have some use for them).
- Add private ctors to inner classes of  `AzureServiceType`.

### CODE REVIEW

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/compatibility/SpringCloudAzureCompatibilityVerifierProperties.java [hunk 1]
```diff
@@ -12,24 +13,44 @@
  */
 @ConfigurationProperties("spring.cloud.azure.compatibility-verifier")
 public class SpringCloudAzureCompatibilityVerifierProperties {
+
+    /**
+     * Whether to enable the Spring Cloud Azure compatibility verifier.
+     */
     private boolean enabled;
+    /**
+     * Comma-delimited list of compatible Spring Boot versions.
```
Comment (jialigit) [inline sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/compatibility/SpringCloudAzureCompatibilityVerifierProperties.java:22]: comment may not correctly describe the field.

Comment (saragluna) [reply sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/compatibility/SpringCloudAzureCompatibilityVerifierProperties.java:22]: Do you have suggestions for this?

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/jms/ServiceBusJmsAutoConfiguration.java [hunk 2]
```diff
@@ -56,10 +57,17 @@ ServiceBusJmsConnectionFactoryCustomizer amqpOpenPropertiesCustomizer() {
...
+     * providers.
+     * @param connectionStringProviders the connection string providers to provide the Service Bus connection string.
+     * @return the bean post processor.
+     */
     @Bean
     @ConditionalOnMissingBean
     @ConditionalOnMissingProperty(prefix = "spring.jms.servicebus", name = "connection-string")
-    public static AzureServiceBusJmsPropertiesBeanPostProcessor azureServiceBusJmsPropertiesBeanPostProcessor(ObjectProvider<ConnectionStringProvider<AzureServiceType.ServiceBus>> connectionStringProviders) {
+    public static AzureServiceBusJmsPropertiesBeanPostProcessor azureServiceBusJmsPropertiesBeanPostProcessor(
+        ObjectProvider<ConnectionStringProvider<AzureServiceType.ServiceBus>> connectionStringProviders) {
```
Comment (jialigit) [inline sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/jms/ServiceBusJmsAutoConfiguration.java]: why arg name use plural style?

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/jms/properties/AzureServiceBusJmsProperties.java [hunk 3]
```diff
@@ -25,19 +25,19 @@
 
     private static final String AMQP_URI_FORMAT = "amqps://%s?amqp.idleTimeout=%d";
     /**
-     * Connection string to connect to service bus.
+     * Connection string to connect to a service bus.
```
Comment (jialigit) [inline sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/jms/properties/AzureServiceBusJmsProperties.java:28]: Service Bus?

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/jms/properties/AzureServiceBusJmsProperties.java [hunk 4]
```diff
@@ -65,70 +65,138 @@
...
+     * @param remoteUrl the URL of the AMQP broker.
+     */
     public void setRemoteUrl(String remoteUrl) {
         this.remoteUrl = remoteUrl;
     }
 
+    /**
+     * Get the login user of the AMQP broker.
+     * @return the login user of the AMQP broker.
+     */
```
Comment (jialigit) [inline sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/jms/properties/AzureServiceBusJmsProperties.java:87]: why need comments in both field and getters at the same time?

Comment (saragluna) [reply sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/jms/properties/AzureServiceBusJmsProperties.java:87]: The architects ask us to do so.

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/jms/ServiceBusJmsAutoConfiguration.java (hunk 2 above)
Comment (saragluna) [reply sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/jms/ServiceBusJmsAutoConfiguration.java]: this is not the change of this pr, but when you use ObjectProvider, it means there could be multiple such objects.

**Review by stliu — APPROVED** [review]

**Pull request merged**